File: 01_Language/01_Functions/php/get_class_vars.php
<?php

class TestCase
{
    public $a    = 1;
    protected $b = 2;
    private $c   = 3;

    public static function expose()
    {
        var_dump(get_class_vars(__CLASS__));
    }
}

TestCase::expose();
echo "=================\n";
var_dump(get_class_vars('TestCase'));

File: 01_Language/01_Functions/python/get_class_vars.py
# coding: utf-8

def get_class_vars(cls):
    return vars(cls)


class TestCase(object):
    d = 1
    _e = 2
    __f = 3

    def __init__(self):
        self.a = 1
        self._b = 2
        self.__c = 3

    @classmethod
    def expose(cls):
        print(get_class_vars(cls))


if __name__ == '__main__':
    TestCase.expose()
    print("================")
    print(get_class_vars(TestCase))
